packages/leadership-hero/main.spec.js

```diff
@@ -0,0 +1,192 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { basicHeroDataAdapter } from '../../global/js/utils';
+import moduleToTest from './main';
+
+const { main } = moduleToTest;
+
+const mockedError = vi.fn();
+console.error = mockedError;
+
+const mockedFetch = {
+    title: 'Leadership test title',
+    pubDateFormatted: 'April 20th, 2025',
+    topic: {
+        asset_name: 'Leadership & Impact',
+        asset_url: 'https://example.com/topic',
+    },
+};
+
+vi.mock('../../global/js/utils', () => ({
+    matrixBasicHeroService: vi.fn(),
+    basicHeroDataAdapter: vi.fn().mockImplementation(() => ({
+        setBasicHeroService: vi.fn(),
+        getBasicHeroData: vi.fn().mockResolvedValue(mockedFetch),
+    })),
+}));
+
+describe('[Leadership Hero]', () => {
+    const defaultMockData = {};
+    const defaultMockInfo = {
+        env: {
+            BASE_DOMAIN: 'https://example.com/api',
+        },
+        fns: {
+            assetId: '162603',
+        },
+    };
+
+    beforeEach(() => {
+        vi.clearAllMocks();
+    });
+
+    // General validation and error handling
+    describe('[Error Handling]', () => {
+        it('Should error when no parameters provided', async () => {
+            const result = await main();
+            expect(result).toContain('<!-- Error occurred in the Leadership Hero component:');
+            expect(mockedError).toBeCalledTimes(1);
+        });
+
+        it('Should throw error when BASE_DOMAIN is missing from env', async () => {
+            const result = await main({}, { env: { BASE_DOMAIN: undefined } });
+            expect(result).toContain('BASE_DOMAIN');
+            expect(mockedError).toBeCalledTimes(1);
+        });
+
+        it('Should throw error when BASE_DOMAIN is missing from set.environment', async () => {
+            const result = await main({}, { set: { environment: { BASE_DOMAIN: undefined } } });
+            expect(result).toContain('BASE_DOMAIN');
+            expect(mockedError).toBeCalledTimes(1);
+        });
+
+        it('Should error if assetId is missing', async () => {
+            const result = await main(defaultMockData, { env: defaultMockInfo.env, fns: { assetId: '' } });
```

Comment:
Use `ctx` instead of `fns`
```suggestion
            const result = await main(defaultMockData, { env: defaultMockInfo.env, ctx: { assetId: '' } });
```

Reply:
Ok, all fns changed into ctx